cdc/sorter/unified/merger.go

```diff
@@ -404,18 +403,40 @@ func runMerger(ctx context.Context, numSorters int, in <-chan *flushTask, out ch
 		return nil
 	}
 
-	resolvedTsNotifierChan := make(chan struct{}, 1)
-	errg, ctx := errgroup.WithContext(ctx)
+	resolvedTsTicker := time.NewTicker(time.Second * 1)
+	resolvedTsTickCh := make(chan struct{})
 
-	errg.Go(func() error {
-		for {
-			var task *flushTask
-			select {
-			case <-ctx.Done():
-				return ctx.Err()
-			case task = <-in:
-			}
+	defer resolvedTsTicker.Stop()
 
+	var lastResolvedTs uint64
+	resolvedTsTickFunc := func() error {
+		curResolvedTs := atomic.LoadUint64(&minResolvedTs)
+		if curResolvedTs > lastResolvedTs {
+			err := onMinResolvedTsUpdate(curResolvedTs)
+			if err != nil {
+				return errors.Trace(err)
+			}
+		} else if curResolvedTs < lastResolvedTs {
+			log.Panic("resolved-ts regressed in sorter",
+				zap.Uint64("curResolved-ts", curResolvedTs),
+				zap.Uint64("lastResolved-ts", lastResolvedTs))
+		}
+		return nil
+	}
+	for {
+		var task *flushTask
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-resolvedTsTicker.C:
+			if err := resolvedTsTickFunc(); err != nil {
+				return err
+			}
+		case <-resolvedTsTickCh:
+			if err := resolvedTsTickFunc(); err != nil {
+				return err
+			}
+		case task = <-in:
```

Comment:
This will reintroduce the problem described in https://github.com/pingcap/tiflow/pull/1574

Reply:
fixed